### PR TITLE
test: Annotations End to End tests

### LINKED ISSUE #849: Annotations End to End test: Do the annotation thing
### Background
This is the overall, high level, end to end test.

### Acceptance Criteria
Things to test:
#### Sahas + Slotman
- Stream work
  - Clicking the x on a stream pill removes that pill from the control bar and stops rendering that stream
    - Searching the stream list when a pill has been removed shows the removed stream
      - Clicking the removed stream adds it
  - Searching the stream list when the pill is active doesn't show it

#### Zoe
- With one click annotations enabled
  - Clicking the graph with the control bar open creates an annotation
  - Clicking the graph with the control bar closed creates an annotations
  - The text written is the text that shows up in the tooltip

#### First come, first served
- Clicking the Annotation button next to the variable button toggles the control bar
- Disabling one click annotations disables adding annotations

### PR DESCRIPTION
**Pull request by ChitlangeSahas**
Closes #849 

Current Progress:

- can create an annotation when the graph is clicked and the control bar is open `✅`
- can create an annotation when graph is clicked and the control bar is closed`✅`
- can hide the Annotations Control bar after clicking on the Annotations Toggle Button `✅`
- can disable writing annotations if Enable-Annotations is disabled `✅`
- can show a tooltip when an annotation is hovered on in the graph `✅`
- can delete an annotation `✅`
- can edit an annotation `✅`
- can create an annotation in a cell and the annotation is scoped to that cell only, it doesn't show in the second cell next to it. `✅`

`🟡` = in progress
`✅` = completed

- [ ] [E2E Tests pass in k8s-idpe](https://docs.influxdata.io/development/guides/local-development/#three-ways-to-run-the-ui-e2e-tests-against-the-kind-cluster) (applicable only if you edited Cypress tests)

### CODE REVIEW

**Comment by hercules-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 43 | :red_circle:
accessibility | 96 | :green_circle:
best-practices | 93 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :question:

File: cypress/e2e/cloud/annotations.test.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    })
+    cy.getByTestID('cell blah').within(() => {
+      cy.getByTestID('giraffe-inner-plot').click()
+    })
+    cy.getByTestID('overlay--container').within(() => {
+      cy.getByTestID('textarea')
+        .should('be.visible')
+        .click()
+        .focused()
+        .type('im a hippopotamus')
```
Comment (hoorayimhelping) [inline cypress/e2e/cloud/annotations.test.ts:61]: https://www.youtube.com/watch?v=mfPCFQfOnLg

File: cypress/e2e/cloud/annotations.test.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+      cy.getByTestID('add-annotation-submit').click()
+    })
+  })
+
+  afterEach(() => {
+    // tests were being flaky because the annotations toggle state would keep leaking into other states
```
Comment (hoorayimhelping) [inline cypress/e2e/cloud/annotations.test.ts:67]: what happened with this? sounds like something we'll need to investigate

File: cypress/e2e/cloud/annotations.test.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+  })
+  it('can hide the stream in the search bar when the stream is active', () => {
+    cy.getByTestID('annotations-search-input')
+      .focus()
+      .click()
+    cy.getByTestID('annotations-searchbar-suggestions').within(() => {
+      cy.getByTestID('list-empty-state').should('be.visible')
+    })
+  })
+  it('can hide the pill and stop rendering the stream once X is clicked on the pill', () => {
```
Comment (hoorayimhelping) [inline cypress/e2e/cloud/annotations.test.ts:78]: :+1:

File: cypress/e2e/cloud/annotations.test.ts [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+          .click()
+      })
+
+    // the rendering should start
+    cy.get('line').should('exist')
+
+    // the pill should be back
+    cy.getByTestID('annotation-pill default').should('exist')
+  })
+  it('can create an annotation when graph is clicked and the control bar is open', () => {
```
Comment (hoorayimhelping) [inline cypress/e2e/cloud/annotations.test.ts:114]:
```suggestion
  it('can create an annotation when the graph is clicked and the control bar is open', () => {
```

File: src/annotations/components/AddAnnotationOverlay.tsx [hunk 5]
```diff
@@ -11,14 +11,19 @@ import {OverlayContext} from 'src/overlays/components/OverlayController'
...
 export const AddAnnotationOverlay: FC = () => {
   const {onClose} = useContext(OverlayContext)
+  const dispatch = useDispatch()
   const {createAnnotation, startTime} = useSelector(getOverlayParams)
 
   const handleSubmit = (modifiedAnnotation): void => {
     const formIsValid = true
     if (formIsValid) {
       createAnnotation(modifiedAnnotation)
+      dispatch(fetchAndSetAnnotationStreams)
```
Comment (hoorayimhelping) [inline src/annotations/components/AddAnnotationOverlay.tsx:26]: isn't it awesome when the tests discover something missing? 🎉

File: yarn.lock [hunk 6]
```diff
@@ -1357,6 +1357,11 @@
     "@types/prop-types" "*"
     csstype "^3.0.2"
 
+"@types/resize-observer-browser@^0.1.5":
```
Comment (hoorayimhelping) [inline yarn.lock:1360]: i don't think this should be here. make sure you checkout this update and merge in the latest master.

**Review by hoorayimhelping — APPROVED** [review]
the descriptions are well done! i did have one question about the afterEach, can you ping me about that on slack please?

**Comment by ChitlangeSahas** [comment]
@daviddkkim testing the slack notifier

**Comment by ChitlangeSahas** [comment]
@daviddkkim testing slack again

File: src/annotations/components/AddAnnotationOverlay.tsx (hunk 5 above)
Comment (hoorayimhelping) [inline src/annotations/components/AddAnnotationOverlay.tsx]: we want to remove this now that streams aren't relevant

**Review by hoorayimhelping — APPROVED** [review]
awesome work on this! a long running team effort we can finally put to bed

**Pull request merged**